internal/auth/oidc/service_callback.go

```diff
@@ -0,0 +1,42 @@
+package oidc
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/kms"
+)
+
+// Callback is an oidc domain service function for processing a successful OIDC
+// Authentication Response from an IdP oidc callback. On success, it returns a
+// final redirect URL for the response to the IdP.
+//
+// For more info on a successful OIDC Authentication Response see:
+// https://openid.net/specs/openid-connect-core-1_0.html#AuthResponse
+//
+// The service operation includes:
+//
+// * Decrypt the state which has been encrypted with the OIDC DEK. If decryption
+// fails, and error is returned. Decrypted state payload includes the
+// token_request_id, nonce and final_redirect_url.
+//
+// * Exchange the callbackCodeParameter for provider tokens and validate the
+// tokens.  Call UserInfo endpoint using access token.
+//
+// * Use oidc.(Repository).upsertAccount to create/update account using ID
+// Tokens claims. The "sub" claim as external ID and setting email and full name
+// for the account.
+//
+// * Use iam.(Repository).LookupUserWithLogin(...) look up the iam.User matching
+// the Account.
+//
+// * Use the authtoken.(Repository).CreateAuthToken(...) to create a pending
+// auth token for the authenticated user.
+func Callback(
+	ctx context.Context,
+	kms *kms.Kms,
```

Comment:
start auth used the KMS associated with OIDC, Shouldn't we do the same here?

Reply:
the function signature has already been updated in the "wip" branch: https://github.com/hashicorp/boundary/blob/jimlambrt-oidc-wip/internal/auth/oidc/service_callback.go